solr/solrj/src/test/org/apache/solr/client/solrj/request/TestCoreAdmin.java

```diff
@@ -46,20 +51,73 @@
 import org.apache.solr.core.SolrCore;
 import org.apache.solr.metrics.SolrCoreMetricManager;
 import org.apache.solr.metrics.SolrMetricManager;
+import org.apache.solr.util.EmbeddedSolrServerTestRule;
 import org.hamcrest.MatcherAssert;
 import org.junit.After;
+import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.RuleChain;
 import org.junit.rules.TestRule;
 
-public class TestCoreAdmin extends AbstractEmbeddedSolrServerTestCase {
+public class TestCoreAdmin extends SolrTestCase {
 
   private static String tempDirProp;
 
+  @Rule public EmbeddedSolrServerTestRule solrClientTestRule = new EmbeddedSolrServerTestRule();
+
+  protected static Path SOLR_HOME;
+  protected static Path CONFIG_HOME;
+
+  protected CoreContainer cores = null;
+
   @Rule public TestRule testRule = RuleChain.outerRule(new SystemPropertiesRestoreRule());
 
+  @BeforeClass
+  public static void setUpHome() throws IOException {
+    // wtf?
+    if (System.getProperty("tempDir") != null) tempDirProp = System.getProperty("tempDir");
+    CONFIG_HOME = getFile("solrj/solr/shared").toPath().toAbsolutePath();
+    SOLR_HOME = createTempDir("solrHome");
+    FileUtils.copyDirectory(CONFIG_HOME.toFile(), SOLR_HOME.toFile());
+  }
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+
+    System.setProperty("solr.solr.home", SOLR_HOME.toString());
+    System.setProperty(
+        "configSetBaseDir", CONFIG_HOME.resolve("../configsets").normalize().toString());
+    System.out.println("Solr home: " + SOLR_HOME.toString());
+
+    // The index is always stored within a temporary directory
+    File tempDir = createTempDir().toFile();
+
+    File dataDir = new File(tempDir, "data1");
+    File dataDir2 = new File(tempDir, "data2");
+    System.setProperty("dataDir1", dataDir.getAbsolutePath());
+    System.setProperty("dataDir2", dataDir2.getAbsolutePath());
+    System.setProperty("tempDir", tempDir.getAbsolutePath());
+    System.setProperty("tests.shardhandler.randomSeed", Long.toString(random().nextLong()));
+    SolrTestCaseJ4.newRandomConfig();
+    solrClientTestRule.build().withSolrHome(SOLR_HOME).init();
+
+    solrClientTestRule
+        .newCollection(DEFAULT_TEST_COLLECTION_NAME)
+        .withConfigSet(CONFIG_HOME.resolve("../configsets").normalize().toString())
```

Comment:
a configSet reference is a simple name, not a path (although it's on disk in a like-named directory).  To know which to use, look in `solr/solrj/src/test-files/solrj/solr/shared/core0/core.properties` which points to one named "shared".